cmd/expiration-mailer/main.go

```diff
@@ -229,16 +231,88 @@ func (m *mailer) findExpiringCertificates() error {
 			m.log.Err(fmt.Sprintf("expiration-mailer: Error loading certificates: %s", err))
 			return err // fatal
 		}
-		if len(certs) > 0 {
-			processingStarted := m.clk.Now()
-			m.processCerts(certs)
-			m.stats.TimingDuration("Mailer.Expiration.ProcessingCertificatesLatency", time.Since(processingStarted), 1.0)
+
+		if len(certs) == 0 {
+			m.log.Info(fmt.Sprintf("expiration-mailer: Found 0 certificates that expire between %s and %s and had last nag >%s before expiry", left, right, expiresIn))
+			continue // nothing to do
 		}
+
+		certs = m.eliminateRenewedCerts(certs)
+
+		processingStarted := m.clk.Now()
+		m.processCerts(certs)
+		m.stats.TimingDuration("Mailer.Expiration.ProcessingCertificatesLatency", time.Since(processingStarted), 1.0)
 	}
 
 	return nil
 }
 
+func (m *mailer) certIsRenewed(serial string) (renewed bool, err error) {
+	present, err := m.dbMap.SelectInt(`
+		SELECT 1
+		FROM fqdnSets a
+		LEFT OUTER JOIN fqdnSets b
+			ON a.setHash = b.setHash
+			AND a.issued < b.issued
+		WHERE a.serial = :serial
+		LIMIT 1`,
+		map[string]interface{}{"serial": serial},
+	)
+	return present == 1, err
+}
+
+// eliminateRenewedCerts returns a new slice with the already-renewed certificates removed.
+func (m *mailer) eliminateRenewedCerts(certs []core.Certificate) []core.Certificate {
+	workCh := make(chan core.Certificate)
+	retCh := make(chan core.Certificate)
+	var wg sync.WaitGroup
+
+	worker := func() {
+		for cert := range workCh {
+			renewed, err := m.certIsRenewed(cert.Serial)
```

Comment:
Why not just do this serially in one of the loops already in `processCerts`? While this is a nice pattern it seems needlessly complex for this implementation.